📝 Update release documentation (#31631)
mistercrunch/Airflow: docs/apache-airflow/release-process.rst

In git, each minor version will have its own branch, called ``vX-Y-stable`` where bugfix/security releases will be issued from.
Commits and PRs should not normally go direct to these branches, but instead should target the main branch and then be cherry-picked by the release managers to these release branches.

> As a general rule of thumb changes that will be included in a bugfix/security release will be associated with the corresponding github milestone in the PR but this is currently a manual process and deviations may occur.

Reviewer: ```suggestion
As a general rule of thumb, changes that will be included in a bugfix/security release will be associated with the corresponding github milestone in the PR but this is currently a manual process and deviations may occur.
```

> In all cases the maintainers reserve the right to punt a PR to a later release if they deem it prudent.

Reviewer 2: ```suggestion
In all cases the release manager reserve the right to punt a PR to a later release if they deem it prudent.
```

Author: resolved in https://github.com/apache/airflow/pull/31631/commits/1a9c7a3ec244e0b4a8c6cc1988d7398e6aa143f5

> In all cases the maintainers reserve the right to punt a PR to a later release if they deem it prudent.

Reviewer: ```suggestion
In all cases, the release managers reserve the right to postpone a PR to a later release if they deem it prudent.
```

Readers like me might not have a strong vocabulary, so suggesting punt -> postpone if you'd like to consider :)
